render/src/lib/views/sidebar.vertical/port.available/components.ts

```diff
@@ -0,0 +1,304 @@
+import { Subscription, Subject } from 'rxjs';
+import Service, { IPort } from '../../../services/service';
+import { EHostEvents } from '../../../common/host.events';
+import { SidebarVerticalPortDialogComponent } from '../port.options/component';
+import { IOptions, CDefaultOptions } from '../../../common/interface.options';
+import { ENotificationType } from 'chipmunk.client.toolkit';
+
+interface Irgb {
+    red: number;
+    green: number;
+    blue: number;
+    opacity: number;
+}
+
+const LIMIT = 500;
+
+@Component({
+    selector: 'lib-dia-port-available-com',
+    templateUrl: './template.html',
+    styleUrls: ['./styles.less']
+})
+
+export class DialogAvailablePortComponent implements OnDestroy, AfterViewInit, OnInit {
+
+    @Input() port: IPortInfo;
+    @Input() observables: {tick: Subject<boolean>, resize: Subject<{ sidebar_width: number, sidebar_height: number }>};
+
+    @ViewChildren('canvas') canvases: QueryList<ElementRef>;
+
+    private _subscriptions: { [key: string]: Subscription } = {};
+    private _canvas: ElementRef<HTMLCanvasElement>;
+    private _ctx: any;
+    private _read: number = 0;
+    private _chart: Chart;
+    private _chart_labels = new Array(LIMIT).fill('');
+    private _destroyed: boolean = false;
+    private _options: IOptions;
+    private _defaultOptions: IOptions = Object.assign({}, CDefaultOptions);
+    private _sidebar_width: number;
+    private _session: string;
+
+    public _ng_isConnected: boolean = false;
+
+    constructor(private _cdRef: ChangeDetectorRef) {
+    }
+
+    ngAfterViewInit() {
+        this._createChart();
+        this._loadSession();
+        this._subscriptions.Subscription = this.observables.tick.subscribe((tick: boolean) => {
+            if (tick) {
+                this._update();
+            }
+            this._forceUpdate();
+        });
+
+        this._subscriptions.Subscription = this.observables.resize.subscribe((size: any) => {
+
+            this._sidebar_width = size.sidebar_width;
+            Service.chart_limit = Math.ceil(this._sidebar_width / 10);
+            this._chart.config.data.labels = this._chart_labels.slice(0, Service.chart_limit);
+
+            const cSessionPort = this._getSessionPort();
+            if (cSessionPort) {
+                this._chart.config.data.datasets[0].data = cSessionPort.sparkline_data.slice(0, Service.chart_limit + 1);
+            } else {
+                console.error('Something went wrong while loading the session');
+            }
+        });
+
+        this._subscriptions.Subscription = Service.getObservable().event.subscribe((message: any) => {
+            if (message.event === EHostEvents.spyState && message.load[this.port.path]) {
+                this._read = message.load[this.port.path];
+            } else if (message.event === EHostEvents.state && message.state[this.port.path]) {
+                this._read = message.state[this.port.path].ioState.read;
+            }
+        });
```

Comment:
1. You've written all subscriptions (from `tick`, `resize` and `event`) into one property `this._subscriptions.Subscription`. It means you will not be able to unsubscribe correctly.
2. I think it's better to move out handlers to methods of call. It would be better